src/layer.js

```diff
@@ -216,4 +228,8 @@ Layer.prototype.draw = function(data) {
 			}
 		}
 	}
+
+	if (!selection.empty()) {
```

Comment:
So I think the way I would have approached it, is by doing the selection check here:
https://github.com/jugglinmike/d3.chart/blob/gh-103-extensions/src/layer.js#L217
and here:
https://github.com/jugglinmike/d3.chart/blob/gh-103-extensions/src/layer.js#L227

What I worry about is the `remove` routine having more destructive behavior (like removing parent containers that the selection would go into) that you might not want to execute unless you're effectively 'done' redrawing this layer or done with the chart itself (or perhaps you want to hide it somehow.)

Is there a reason we couldn't localize it to the specific lifecycle events and just not call the handlers?